core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4548,22 +4562,59 @@ private void analyzeRowFilter(String currentIdentity, Table table, QualifiedObje
                 throw new TrinoException(INVALID_ROW_FILTER, extractLocation(table), format("Row filter for '%s' is recursive", name), null);
             }
 
-            Expression expression;
-            try {
-                expression = sqlParser.createExpression(filter.getExpression(), createParsingOptions(session));
-            }
-            catch (ParsingException e) {
-                throw new TrinoException(INVALID_ROW_FILTER, extractLocation(table), format("Invalid row filter for '%s': %s", name, e.getErrorMessage()), e);
-            }
+            Expression expression = parseRowConstraintExpression(table, name, filter, INVALID_ROW_FILTER, "Invalid row filter");
```

Comment:
Revert this. This is spurious reuse of functionality. We don't need to add an abstraction for a "row constraint expression". It's a bit of a stretch to call row filters and check constraints "constraint expressions"